usdt-impl/build.rs

```diff
@@ -27,20 +27,46 @@ enum Backend {
     NoOp,
 }
 
+// Extract any -C ARGS ... passed to RUSTFLAGS
+fn extract_encoded_rustc_flags() -> Vec<String> {
```

Comment:
This function returns _all_ `-C` flag values, however the only consumption site filters that down to a flag that is `link-dead-code`. It would be simpler I think to have this function do all that work itself, returning `true` if `link-dead-code` is specified with a value other than `n`.